tools/updateRollup/updateRollup.ts

```diff
@@ -85,31 +92,137 @@ async function main() {
 
+    const { polygonRollupManagerAddress } =  updateRollupsParameters;
+
     // Load Rollup manager
     const PolgonRollupManagerFactory = await ethers.getContractFactory("PolygonRollupManager", deployer);
     const rollupManagerContract = PolgonRollupManagerFactory.attach(
         polygonRollupManagerAddress
     ) as PolygonRollupManager;
 
-    // Check role
-    const UPDATE_ROLLUP_ROLE = ethers.id("UPDATE_ROLLUP_ROLE");
+    const outputsJson = [];
+
+    for(let i = 0; i < updateRollupsParameters.rollups.length; i++) {
+        const outputJson = {} as any;
+        const updateRollupParameters = updateRollupsParameters.rollups[i];
+
+        /*
+         * Check parameters
+         * Check that every necessary parameter is fullfilled
+         */
+        const mandatoryParametersRollup = [
+            "rollupAddress",
+            "newRollupTypeID",
+            "upgradeData",
+        ];
+
+        for (const parameterName of mandatoryParametersRollup) {
+            if (updateRollupParameters[parameterName] === undefined || updateRollupParameters[parameterName] === "") {
+                throw new Error(`Missing rollup[${i}] parameter: ${parameterName}`);
+            }
+        }
 
-    if ((await rollupManagerContract.hasRole(UPDATE_ROLLUP_ROLE, deployer.address)) == false) {
-        // log that address ha sno role
-        throw new Error(`Address ${deployer.address} does not have the UPDATE_ROLLUP_ROLE role`);
-    }
+        const {rollupAddress, newRollupTypeID, upgradeData} = updateRollupParameters;
 
-    // Add a new rollup type with timelock
-    console.log(await (await rollupManagerContract.updateRollup(rollupAddress, newRollupTypeID, upgradeData)).wait());
+        outputJson.networkName = network.name;
+        outputJson.polygonRollupManagerAddress = polygonRollupManagerAddress;
+        outputJson.rollupAddress = rollupAddress;
+        outputJson.newRollupTypeID = newRollupTypeID;
+        outputJson.upgradeData = upgradeData;
 
-    outputJson.networkName = network.name;
-    outputJson.polygonRollupManagerAddress = polygonRollupManagerAddress;
-    outputJson.rollupAddress = rollupAddress;
-    outputJson.newRollupTypeID = newRollupTypeID;
-    outputJson.upgradeData = upgradeData;
+        if(updateRollupsParameters.type === transactionTypes.EOA) {
+            // Check role
+            const UPDATE_ROLLUP_ROLE = ethers.id("UPDATE_ROLLUP_ROLE");
+            if ((await rollupManagerContract.hasRole(UPDATE_ROLLUP_ROLE, deployer.address)) == false) {
+                // log that address has no role
+                throw new Error(`Address ${deployer.address} does not have the UPDATE_ROLLUP_ROLE role`);
+            }
+            console.log(`Updating rollup ${rollupAddress}...`)
+            console.log(await (await rollupManagerContract.updateRollup(rollupAddress, newRollupTypeID, upgradeData)).wait());
+        } else if(updateRollupsParameters.type === transactionTypes.TIMELOCK) {
```

Comment:
we cna schedule batch for those cases, check if there are more than one tx.timelock